src/devices/Mcp3xxx/Mcp30xx32xx.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Spi;
+
+namespace Iot.Device.Adc
+{
+    /// <summary>
+    /// Mcp30xx32xx Abstract class representing the MCP 10 and 12 bit ADC devices.
+    /// </summary>
+    public abstract class Mcp30xx32xx : Mcp3xxx
+    {
+        /// <summary>
+        /// the number of input pins on the ADC
+        /// </summary>
+        protected byte PinCount;
+
+        private byte _adcResolutionBits;
+
+        /// <summary>
+        /// Constructs Mcp30xx32xx instance
+        /// </summary>
+        /// <param name="spiDevice">Device used for SPI communication</param>
+        /// <param name="pinCount">Value representing the number of pins on the device.</param>
+        /// <param name="adcResolutionBits">The number of bits of resolution for the ADC.</param>
+        public Mcp30xx32xx(SpiDevice spiDevice, byte pinCount, byte adcResolutionBits) : base(spiDevice)
```

Comment:
Does pinCount refer to number of ADC channels? If so please rename to `channelCount` (here and everywhere where applies) - if not please clarify docs

Reply:
I wanted to distinguish between the word channel and pins as in the earlier api then channel could refer to a channel pair which there were pins/2 of. Now that the api has changed then this can be changed.